EKF/RingBuffer.h

```diff
@@ -63,12 +57,20 @@ class RingBuffer
 
 	bool allocate(uint8_t size)
 	{
+		if (valid() && (size == _size)) {
+			// no change
+			return true;
+		}
+
+		if (size == 0) {
+			return false;
+		}
 
 		if (_buffer != nullptr) {
 			delete[] _buffer;
 		}
 
-		_buffer = new data_type[size];
+		_buffer = new data_type[size]{};
```

Comment:
I guess it's just once, but we shouldn't be using the data anyways. For these kind of things I like to initialize with 0xDEADBEEF or something similar, at least for debug builds.

Reply:
What I discovered in https://github.com/PX4/PX4-ECL/pull/935 (the larger cleanup this was broken out of) is that there seems to be a few minor (hopefully) cases where it matters. I'd still like to fix that entirely and really ensure we're not using anything before there's valid data, but it needs some debug time.